### PR TITLE
chore(config): stack + data subject

### PR DESCRIPTION
**Pull request by smalexan**
## Description of this change

Add config stacks and mapping from purpose to stack
Add config data subjects and mapping to rights and purposes
Add theme secondary button color
Add experience configurations + extensions

## Why is this change being made?
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [x] I have informed stakeholders of my changes.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -238,7 +237,12 @@ export interface Purpose {
   canonicalPurposeCode?: string
   legalBasisName?: string
   legalBasisDescription?: string
+
+  // the identifier of the stack that the purpose belongs to when displayed. If empty, this purpose belongs to no stack
```
Comment (sethyates) [inline src/index.ts:241]: this feels back to front. wouldn't the stack list the purpose codes in order?

File: src/index.ts [hunk 2]
```diff
@@ -238,7 +237,12 @@ export interface Purpose {
...
   stackID?: string
+
+  // the data subjects for which the purpose is relevant. If this list is empty, the purpose applies to all data subjects
+  dataSubjectCodes?: string[]
```
Comment (sethyates) [inline src/index.ts:245]: shouldn't we add this to rights too?

Comment (sethyates) [reply src/index.ts:245]: these are actually `dataSubjectTypes`

File: src/index.ts [hunk 3]
```diff
@@ -385,11 +392,13 @@ export interface Vendor {
   UsesNonCookieAccess?: boolean
```
Comment (sethyates) [inline src/index.ts:392]: is this really the right casing for this?

File: src/index.ts (hunk 1 above)
Comment (smalexan) [reply src/index.ts:241]: Yeah that works. Then keep track if a purpose is not a member of a stack and display those below the stack?

File: src/index.ts (hunk 2 above)
Comment (smalexan) [reply src/index.ts]: Updated to dataSubjectTypes

File: src/index.ts (hunk 3 above)
Comment (smalexan) [reply src/index.ts]: nope added correct and marked this as deprecated

File: src/index.ts [hunk 4]
```diff
@@ -379,7 +435,25 @@ export interface Vendor {
...
+}
+
+// DataSubjectType represents user defined data subject types with code as the unique identifier
+export interface DataSubjectType {
+  code: string
+  name: string
+}
+
+// Stack represents a grouping of purposes to be displayed in an experience
+export interface Stack {
```
Comment (sethyates) [inline src/index.ts:451]: no code?

Comment (smalexan) [reply src/index.ts:451]: Could have it but FE should not need it

**Review by sethyates — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (sethyates) [reply src/index.ts]: Only those listed in stacks are shown, right? Or is the definition that any not in a stack are shown at a different level? Or do we have a "default" stack?

**Pull request merged**